### ISSUE TITLE
[Machine learning] Automated hyperparameter tuning using genetic algorithms

### ISSUE THREAD

**Issue by donnexwafula33** [section-engineering-education/engineering-education#7030]
#### Proposed title of article
[Machine learning] Automated hyperparameter tuning using genetic algorithms

#### Proposed article introduction
A hyperparameter is a parameter that is set before the learning process begins. These parameters are tunable and can directly affect how well the model trains. Some examples of hyperparameters in machine learning: Learning Rate, number of Epochs, momentum, regularization constant, number of branches in a decision tree, and number of clusters in a clustering algorithm.

Some common Hyperparameter tuning methods and techniques are:
- Grid Search: Search a set of manually predefined hyperparameters for the best performing hyperparameter. 
- Random Search: Similar to grid search, but replaces the exhaustive search with random search. 
- Bayesian Optimization: Builds a probabilistic model of the function mapping from hyperparameter values to the target evaluated on a validation set. 
- Gradient-Based Optimization: Compute gradient using hyperparameters and then optimize hyperparameters using gradient descent.
- Genetic algorithms: Uses evolutionary algorithms (e.g. genetic functions) to search the space of possible hyperparameters. 

In this tutorial, we will focus on the use of genetic algorithms. Genetic algorithms make use of basic concepts from evolution by natural selection to optimize arbitrary functions. The genetic algorithm that we will implement is TPOT. TPOT's optimization algorithm is stochastic in nature, which means that it uses randomness (in part) to search the possible pipeline space.

#### Key takeaways 
1. Parameter vs hyperparameter in machine learning models.
2. Hyperparameter tuning methods and techniques (Grid search, random search, Optuna, Bayesian optimization, Genetic algorithms)
3. Concepts of natural selection and survival of the fittest.
4. Importing TPOT Evolution Algorithm.
5. Setting the `generations` , `population size` and `offspring` values.
6. Automatic optimization using TPOT pipeline.
#### Article quality
This article explains the common strategies for optimizing hyperparameters (Grid search, random search, Optuna, Bayesian optimization, and genetic algorithms) in detail. A reader will be able to differentiate these strategies. We will implement TPOT which is a genetic algorithm for optimization. We will use TPOT to automatically create multiple pipelines to find the best parameters. We will the compare the accuracy score of model before and after hyperparameter tuning.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)
